kratos/processes/levelset_convection_process.h

```diff
@@ -225,19 +248,76 @@ class LevelSetConvectionProcess
+                    auto it_node = mpDistanceModelPart->NodesBegin() + i_node;
 
-                const array_1d<double,3>& v = mVelocity[i_node];
-                const array_1d<double,3>& v_old = mVelocityOld[i_node];
+                    const array_1d<double,3>& v = mVelocity[i_node];
+                    const array_1d<double,3>& v_old = mVelocityOld[i_node];
 
-                it_node->FastGetSolutionStepValue(VELOCITY) = Nold * v_old + Nnew * v;
-                it_node->FastGetSolutionStepValue(VELOCITY, 1) = Nold_before * v_old + Nnew_before * v;
-                it_node->FastGetSolutionStepValue(mrLevelSetVar, 1) = it_node->FastGetSolutionStepValue(mrLevelSetVar);
+                    it_node->FastGetSolutionStepValue(mrConvectVar) = Nold * v_old + Nnew * v;
+                    it_node->FastGetSolutionStepValue(mrConvectVar, 1) = Nold_before * v_old + Nnew_before * v;
+                    it_node->FastGetSolutionStepValue(mrLevelSetVar, 1) = it_node->FastGetSolutionStepValue(mrLevelSetVar);
+                }
             }
+            else{
+                #pragma omp parallel for
+                for (int i_node = 0; i_node < static_cast<int>(mpDistanceModelPart->NumberOfNodes()); ++i_node){
+                    auto it_node = mpDistanceModelPart->NodesBegin() + i_node;
+
+                    const array_1d<double,3>& v = mVelocity[i_node];
+                    const array_1d<double,3>& v_old = mVelocityOld[i_node];
+
+                    it_node->FastGetSolutionStepValue(mrConvectVar) = Nold * v_old + Nnew * v;
+                    it_node->FastGetSolutionStepValue(mrConvectVar, 1) = Nold_before * v_old + Nnew_before * v;
+                    const double dist = it_node->FastGetSolutionStepValue(mrLevelSetVar);
+                    it_node->FastGetSolutionStepValue(mrLevelSetVar, 1) = dist;
+                    it_node->SetValue(mrLevelSetVar, dist);
+                }
+            }
+
+            mpSolvingStrategy->Solve(); // forward convection to reach phi_n+1
+
+            if (mBfeccOrder > 0) {// Error Compensation and Correction
+                #pragma omp parallel for
+                for (int i_node = 0; i_node < static_cast<int>(mpDistanceModelPart->NumberOfNodes()); ++i_node){
+                    auto it_node = mpDistanceModelPart->NodesBegin() + i_node;
+
+                    const array_1d<double,3>& v = mVelocity[i_node];
+                    const array_1d<double,3>& v_old = mVelocityOld[i_node];
+
+                    it_node->FastGetSolutionStepValue(mrConvectVar) = -(Nold * v_old + Nnew * v);
+                    it_node->FastGetSolutionStepValue(mrConvectVar, 1) = -(Nold_before * v_old + Nnew_before * v);
+                    it_node->FastGetSolutionStepValue(mrLevelSetVar, 1) = it_node->FastGetSolutionStepValue(mrLevelSetVar);
+                }
 
-            mpSolvingStrategy->Solve();
+                mpSolvingStrategy->Solve(); // backward convetion to obtain phi_n*
```

Comment:
`Solve` is deprecated. `InitializeSolutionStep`, `SolveSolutionStep` ... must be called instead.

Reply:
(same for the rest).

Reply:
I assume I should use "SolveSolutionStep", correct?

Reply:
Yes. In the `Initialize` method of the process I'd call the `Initialize` of the strategy (same for the `Finalize`). Then we need to call the `InitializeSolutionStep`,`Predict`, `SolveSolutionStep` and `FinalizeSolutionStep` when we need to solve. See [here](https://github.com/KratosMultiphysics/Kratos/blob/6974c9091ea73741a1a784a4d971064669a888ea/kratos/solving_strategies/strategies/solving_strategy.h#L195-L205)